File: MonkeyJump/ViewControllers/GameViewController.m
//
//  ViewController.m
//  MonkeyJump
//
//  Created by Kauserali on 17/11/13.
//  Copyright (c) 2013 Raywenderlich. All rights reserved.
//

#import "GameViewController.h"
#import "GameScene.h"
#import "GameOverViewController.h"

@interface GameViewController()<GameSceneProtocol>

@end

@implementation GameViewController

- (void)viewWillLayoutSubviews
{
    [super viewWillLayoutSubviews];

    SKView * skView = (SKView *)self.view;
    if (!skView.scene) {
        skView.showsFPS = YES;
        skView.showsNodeCount = YES;
        skView.ignoresSiblingOrder = YES;
        
        GameScene * scene = [GameScene sceneWithSize:skView.bounds.size];
        scene.gameSceneDelegate = self;
        scene.scaleMode = SKSceneScaleModeAspectFill;
        
        [skView presentScene:scene];
    }
}

#pragma mark GameScene delegate

- (void)gameOverWithScore:(int)score
{
    GameOverViewController *gameOverViewController = [self.storyboard instantiateViewControllerWithIdentifier:@"GameOverViewController"];
    [gameOverViewController setScore:score];
    [self.navigationController pushViewController:gameOverViewController animated:NO];
}
@end

File: MonkeyJump/ViewControllers/GameViewController.swift
//
//  GameViewController.swift
//  MonkeyJump
//
//  Created by philippe eggel on 22/12/2015.
//  Copyright © 2015 PhilEagleDev.com. All rights reserved.
//

import SpriteKit

class GameViewController: UIViewController {

    static let storyboardID = "GameViewController"
    var gameTrackRecord: GameTrackRecord?
    
    override func viewWillLayoutSubviews() {
        super.viewWillLayoutSubviews()
        
        let skView = (view as! SKView)
        
        if skView.scene == nil {
            skView.showsFPS = true
            skView.showsNodeCount = true
            skView.ignoresSiblingOrder = true
            
            let scene = GameScene(size: skView.bounds.size)
            scene.gameSceneDelegate = self
            scene.scaleMode = .AspectFill
            scene.challengerGameTrackRecord = gameTrackRecord
            
            skView.presentScene(scene)
        }

    }
}

extension GameViewController: GameSceneProtocol {
    
    func gameOverWithScore(score: Int64, gameTrackRecord: GameTrackRecord) {
        let gameOverViewController = storyboard!.instantiateViewControllerWithIdentifier(GameOverViewController.storyboardID) as! GameOverViewController
        gameOverViewController.setScore(score, gameTrackRecord: gameTrackRecord)
        navigationController?.pushViewController(gameOverViewController, animated: false)
    }
    
}